### ISSUE TITLE
Tensorflow model is fine, but the converted mlmodel gave me wrong answer

### ISSUE THREAD

**Issue by xzhub** [tf-coreml/tf-coreml#201]
Hers is how to reproduce:
1. Retrain to get a new model
```
python -m scripts.retrain --bottleneck_dir=flowers/bottlenecks \
   --model_dir=flowers/models/ \
   --summaries_dir=flowers/training_summaries/mobilenet_1.0_224_lr_0.1_8000 \
   --output_graph=flowers/retrained_graph_0.1_8000.pb \
   --output_labels=flowers/retrained_labels_0.1_8000.txt \
   --architecture=mobilenet_1.0_224 --learning_rate=0.1 \
   --how_many_training_steps=8000 --image_dir=flower_photos
```

2. Check and make sure the model is working:
```
$ python -m scripts.label_image --graph=flowers/retrained_graph_0.1_8000.pb \
   --labels=flowers/retrained_labels_0.1_8000.txt \
   --image=flower_photos/daisy/21652746_cc379e0eea_m.jpg
2018-06-10 09:45:33.848636: I tensorflow/core/platform/cpu_feature_guard.cc:140] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA

Evaluation time (1-image): 0.122s

daisy (score=1.00000)
dandelion (score=0.00000)
sunflowers (score=0.00000)
tulips (score=0.00000)
roses (score=0.00000)
```

3. Convert the model graph to core ml:
```
python my_converter.py flowers/retrained_graph_0.1_8000.pb \
                        flowers/retrained_labels_0.1_8000.txt \
                        flower.mlmodel
```
While the converter is:
```
$ cat my_converter.py
import sys
import tfcoreml as tf_converter
tf_converter.convert(tf_model_path = sys.argv[1],
                     class_labels = sys.argv[2],
                     mlmodel_path = sys.argv[3],
                     output_feature_names = ['final_result:0'],
                     image_input_names = ['input:0']
                     )
````
4. After get the Core ML model flower.mlmodel, use it in iOS. I just cloned the Car Recognition app (https://github.com/likedan/Core-ML-Car-Recognition), replace the mlmodel file with flower.mlmodel. Here is the result I got:
![img_2234](https://user-images.githubusercontent.com/3990687/41204023-1cf9dc4a-6c94-11e8-8d4d-899a85f65ecf.PNG)

**Comment by aseemw** [comment]
Did you set the pre-processing parameters properly? See [notebook](https://github.com/tf-coreml/tf-coreml/blob/master/examples/inception_v1_preprocessing_steps.ipynb).

**Comment by xzhub** [comment]
Maybe not, but I am not fully understand the statement in that notebook "the output of the Softmax op towards the end of the graph is the output ("InceptionV1/Logits/Predictions/Softmax:0"). Lets convert to mlmodel now."

Here is the output of the graph "flowers/retrained_graph_0.1_8000.pb". What should be the parameters in tf_converter.convert?

```
op id 0 : op type: "Placeholder"
input(s):

output(s):
name = input:0, shape: (1, 224, 224, 3),


op id 1 : op type: "Const"
input(s):

output(s):
name = MobilenetV1/Conv2d_0/weights:0, shape: (3, 3, 3, 32),


op id 2 : op type: "Identity"
input(s):
name = MobilenetV1/Conv2d_0/weights:0, shape: (3, 3, 3, 32),

output(s):
name = MobilenetV1/Conv2d_0/weights/read:0, shape: (3, 3, 3, 32),


op id 557 : op type: "MatMul"
input(s):
name = input_1/BottleneckInputPlaceholder:0, shape: (?, 1001),
name = final_training_ops/weights/final_weights/read:0, shape: (1001, 5),

output(s):
name = final_training_ops/Wx_plus_b/MatMul:0, shape: (?, 5),


op id 558 : op type: "Add"
input(s):
name = final_training_ops/Wx_plus_b/MatMul:0, shape: (?, 5),
name = final_training_ops/biases/final_biases/read:0, shape: (5,),

output(s):
name = final_training_ops/Wx_plus_b/add:0, shape: (?, 5),


op id 559 : op type: "Softmax"
input(s):
name = final_training_ops/Wx_plus_b/add:0, shape: (?, 5),

output(s):
name = final_result:0, shape: (?, 5),
```

**Comment by xzhub** [comment]
Let me upload the graph file for your convenience. 

retrained_graph_0.1_8000.pb and its label:
[Archive.zip](https://github.com/tf-coreml/tf-coreml/files/2088103/Archive.zip)

**Comment by xzhub** [comment]
Keep reading the notebook, I think I found the answer in the end. I changed the converter to 
```
import sys
import tfcoreml as tf_converter
tf_converter.convert(tf_model_path = sys.argv[1],
                     mlmodel_path = sys.argv[3],
                     output_feature_names = ['final_result:0'],
                     class_labels = sys.argv[2],
                     image_input_names = ['input:0'],
                     input_name_shape_dict = {"input:0":[1,224,224,3]},
                     red_bias = -1,
                     green_bias = -1,
                     blue_bias = -1,
                     image_scale = 2.0/255.0
                     )
```

For now, at least the iOS app tells me this flow is a daisy. 

I would suggest to put part of this notebook content into README.md, since the ipynb can not be viewed directly online.
